Gems/LmbrCentral/Code/Tests/EditorShapeComponentIntersectionTests.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzManipulatorTestFramework/AzManipulatorTestFrameworkTestHelpers.h>
+#include <AzTest/AzTest.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+#include <AzToolsFramework/Viewport/ViewportSettings.h>
+#include <AzToolsFramework/Viewport/ViewportTypes.h>
+#include <AzToolsFramework/ViewportSelection/EditorSelectionUtil.h>
+
+#include "Shape/EditorSphereShapeComponent.h"
+
+namespace LmbrCentral
+{
+    using AzToolsFramework::ViewportInteraction::BuildMouseButtons;
+    using AzToolsFramework::ViewportInteraction::BuildMouseInteraction;
+    using AzToolsFramework::ViewportInteraction::BuildMousePick;
+
+    class EditorSphereShapeComponentFixture : public UnitTest::ToolsApplicationFixture
+    {
+    public:
+        void SetUpEditorFixtureImpl() override
+        {
+            AZ::SerializeContext* serializeContext = nullptr;
+            AZ::ComponentApplicationBus::BroadcastResult(serializeContext, &AZ::ComponentApplicationBus::Events::GetSerializeContext);
+
+            m_editorSphereShapeComponentDescriptor =
+                AZStd::unique_ptr<AZ::ComponentDescriptor>(EditorSphereShapeComponent::CreateDescriptor());
+            m_editorSphereShapeComponentDescriptor->Reflect(serializeContext);
+
+            m_entityId1 = UnitTest::CreateDefaultEditorEntity("Entity1");
+        }
+
+        void TearDownEditorFixtureImpl() override
+        {
+            bool entityDestroyed = false;
+            AzToolsFramework::EditorEntityContextRequestBus::BroadcastResult(
+                entityDestroyed, &AzToolsFramework::EditorEntityContextRequestBus::Events::DestroyEditorEntity, m_entityId1);
+
+            m_editorSphereShapeComponentDescriptor.reset();
+        }
+
+        AZ::EntityId m_entityId1;
+        AZStd::unique_ptr<AZ::ComponentDescriptor> m_editorSphereShapeComponentDescriptor;
+    };
+
+    struct IntersectionQueryOutcome
+    {
+        bool m_helpersVisible;
+        bool m_expectedIntersection;
+    };
+
+    using ShapeComponentIndirectCallManipulatorViewportInteractionFixture =
+        UnitTest::IndirectCallManipulatorViewportInteractionFixtureMixin<EditorSphereShapeComponentFixture>;
+
+    class ShapeComponentIndirectCallManipulatorViewportInteractionFixtureParam
+        : public ShapeComponentIndirectCallManipulatorViewportInteractionFixture
+        , public ::testing::WithParamInterface<IntersectionQueryOutcome>
+    {
+    public:
+        void SetUpEditorFixtureImpl() override
+        {
+            ShapeComponentIndirectCallManipulatorViewportInteractionFixture::SetUpEditorFixtureImpl();
+
+            auto* entity1 = AzToolsFramework::GetEntityById(m_entityId1);
+            entity1->Deactivate();
```

Comment:
I can see useful here to ASSERT that `entity1` is not `nullptr`, in case the base class changes in the future and child classes are missed to be updated (unlikely, but it could happen and it'll be easier to detect).

Reply:
Hmm you could be right, but my philosophy is that is not what the test is verifying, so I don't really want that mixed in here (there should, and I hope is, a test to verify this elsewhere to ensure it doesn't break). For that reason I'd rather leave it for now (or possibly add an `AZ_Assert` as a sanity check, but not an actual test assert)

Reply:
This is now updated :+1: